File: leetcode-questions/csharp/src/test/csharp/com/leetcode/questions/_3sum/_3sumTest.cs
using Xunit;
using System.Collections.Generic;
using com.leetcode.questions.leetcode.editor.en._3sum;
using com.leetcode.questions.support;

namespace com.leetcode.questions._3sum;

public class _3sumTest {

    private const string CasesFileName = "3sum.json";

    public _3sumTest() {

    }

    [Fact]
    public void TestCases() {
        GeneratedLeetCodeTestHarness.Run(typeof(Solution), "ThreeSum", GeneratedLeetCodeTestHarness.ReadCasesFile(CasesFileName));
    }
}

File: leetcode-questions/java/src/test/java/com/leetcode/questions/_3sum/_3sumTest.java
package com.leetcode.questions._3sum;

import com.leetcode.questions.leetcode.editor.en._3sum;
import com.leetcode.questions.support.GeneratedLeetCodeTestHarness;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.nio.file.Files;
import java.nio.file.Path;

public class _3sumTest {

    private static final Path CASES_PATH = Path.of("generated-test-cases", "3sum.json");

    public _3sumTest() {

    }

    @Test
    public void testCases() throws Exception {
        GeneratedLeetCodeTestHarness.run(_3sum.Solution.class, "solve", Files.readString(CASES_PATH));
    }
}
